### ISSUE TITLE
False-positive 'const-param' warning

### ISSUE THREAD

**Issue by imorda** [tbricks/itiviti-cpp-analyzer#8]
## Description
When a member function tries to modify some fields of a dereferenced field of its argument passed by non-const reference, we get a `[const-param]` warning that suggests to make such argument `const`. 

## Example code
```c++
struct Node
{
    void update(Node & parent) // no warning expected
    {
        if (parent.left != nullptr) {
            parent.left->parent = &parent;
        }
    }

    Node * left;
    Node * right;
    Node * parent;
};
```

## Compile arguments
```
[ 50%] Building CXX object CMakeFiles/ica_test.dir/main.cpp.o
/usr/bin/c++    -g   -Xclang -plugin-arg-ica-plugin -Xclang checks=all=err -Xclang -load -Xclang /usr/local/lib/libica-plugin.so -Xclang -add-plugin -Xclang ica-plugin -std=gnu++17 -o CMakeFiles/ica_test.dir/main.cpp.o -c /home/imodre/CLionProjects/ica-test/main.cpp
/home/imodre/CLionProjects/ica-test/main.cpp:3:17: error: 'parent' can have 'const' qualifier [const-param]
    void update(Node & parent) // no warning expected
                ^
1 error generated.
```

## Build info
Default build from master branch of this repo, clang 10
